tests/stress/deployments.py

```diff
@@ -0,0 +1,761 @@
+
+                    if pod_status == "Running" and container_status == "true":
+                        print(f"Pod {pod_name} is running and ready!")
+                        break
+                    else:
+                        print(
+                            f"Pod {pod_name} not ready yet, status: {pod_status}, container ready: {container_status}"
+                        )
+                        time.sleep(5)
+
+                except Exception as e:
+                    print(f"Pod not ready yet!")
+                    time.sleep(1)
+
+        print("All pods ready!")
+
+    def execute(self, query: str) -> None:
+        if self._querying_type == "data":
+            self._execute_on_data_instance(query)
+        elif self._querying_type == "bolt+routing":
+            self._execute_bolt_routing_query(query)
+
+    def execute_and_fetch(self, query: str) -> None:
+        if self._querying_type == "data":
+            return self._execute_and_fetch_on_data_instance(query)
+
+        raise Exception("Execute and fetch not applicable at the moment on bolt+routing query!")
+
+    def _execute_on_data_instance(self, query: str):
+        """Execute a Cypher query on Memgraph"""
+        # Get Minikube IP and NodePort
+        minikube_ip = self.get_minikube_ip()
+        if not minikube_ip:
+            print("Failed to get Minikube IP.")
+            return
+
+        nodeport = self.get_service_nodeport(self._data_ext_service_names[0])
+        if not nodeport:
+            print("Failed to get NodePort for Memgraph service.")
+            return
+
+        # Connect to Memgraph using the Minikube IP and NodePort
+        memgraph = Memgraph(host=minikube_ip, port=nodeport)
+
+        # Execute the Cypher query
+        memgraph.execute(query)
+
+    def _execute_and_fetch_on_data_instance(self, query: str):
+        """Execute a Cypher query on Memgraph"""
+        # Get Minikube IP and NodePort
+        minikube_ip = self.get_minikube_ip()
+        if not minikube_ip:
+            print("Failed to get Minikube IP.")
+            return
+
+        nodeport = self.get_service_nodeport(self._data_ext_service_names[0])
+        if not nodeport:
+            print("Failed to get NodePort for Memgraph service.")
+            return
+
+        # Connect to Memgraph using the Minikube IP and NodePort
+        memgraph = Memgraph(host=minikube_ip, port=nodeport)
+
+        # Execute the Cypher query
+        return memgraph.execute_and_fetch(query)
+
+    def _execute_bolt_routing_query(self, query: str):
+        minikube_ip = self.get_minikube_ip()
+        service_nodeports = [self.get_service_nodeport(x) for x in self._coord_ext_service_names]
+        successful_query = False
+        for nodeport in service_nodeports:
+            try:
+                uri = f"neo4j://{minikube_ip}:{nodeport}"
+                driver = GraphDatabase.driver(uri, auth=("", ""))
```

Comment:
BDO uses authentication... Just a point you will want to add this at some point in the future.